pseudopotential_calculator/calculations/slab.py

```diff
@@ -34,6 +34,45 @@ def kpoint_mp_grid(self: Self) -> str:
         return f"{self.n_k_points} {self.n_k_points} {1}"
 
 
+def repeat_slab(slab: Atoms, amount: tuple[int, int, int]) -> Atoms:
+    return slab.repeat(amount)  # type: ignore bad lib
+
+
+def add_atom_onto_slab(atom: Atom, slab: Atoms) -> Atoms:
+    slab.append(atom)  # type: ignore bad lib
+    return slab
+
+
+def get_top_position(atom: Atoms) -> list[float]:
+    positions = atom.positions  # type: ignore bad lib
+    return max(positions, key=lambda x: x[2])  # type: ignore bad lib
+
+
+def get_nxn_repeats(atom: Atoms) -> int:
+    n_slab_atoms = len(atom.positions) - 1  # type: ignore bad lib
+    n_atoms_per_repeat = get_n_slab_layer(atom) - 1
+    return np.sqrt(round(n_slab_atoms / n_atoms_per_repeat))
+
+
+def plot_energy_against_nxn_slab(
```

Comment:
better to calculate slab shape in x,y and take the minimum. Would avoid the dodgy np.sqrt

Reply:
I'm not sure how to do this generally